src/System.Threading.Channels/tests/Stress.cs

```diff
@@ -0,0 +1,144 @@
+            int writerTasksCount;
+
+            if (options.SingleReader)
+            {
+                readerTasksCount = 1;
+                writerTasksCount = options.SingleWriter ? 1 : MaxTaskCounts - 1;
+            }
+            else if (options.SingleWriter)
+            {
+                writerTasksCount = 1;
+                readerTasksCount = MaxTaskCounts - 1;
+            }
+            else
+            {
+                readerTasksCount = MaxTaskCounts / 2;
+                writerTasksCount = MaxTaskCounts - readerTasksCount;
+            }
+
+            int readCount = 0;
+
+            for (int i=0; i < readerTasksCount; i++)
+            {
+                taskList.Add(Task.Run(async delegate
+                {
+                    while (true)
+                    {
+                        try
+                        {
+                            if (!reader.TryRead(out int value))
+                            {
+                                value = await reader.ReadAsync();
+                            }
+                            Interlocked.Increment(ref readCount);
+                        }
+                        catch (ChannelClosedException)
+                        {
+                            break;
+                        }
+                    }
+                }));
+            }
+
+            int numberToWriteToQueue = -1;
+            int remainingWriters = writerTasksCount;
+
+            for (int i=0; i < writerTasksCount; i++)
+            {
+                taskList.Add(Task.Run(async delegate
+                {
+                    int num = Interlocked.Increment(ref numberToWriteToQueue);
+                    while (num < MaxNumberToWriteToChannel)
+                    {
+                        await writer.WriteAsync(num);
+                        num = Interlocked.Increment(ref numberToWriteToQueue);
+                    }
+
+                    if (Interlocked.Decrement(ref remainingWriters) == 0)
+                        writer.Complete();
+                }));
+            }
+
+            Task.WaitAll(taskList.ToArray());
+
+            Assert.True(readCount <= MaxNumberToWriteToChannel);
+
+            if (shouldReadAllWrittenValues)
+            {
+                Assert.Equal(MaxNumberToWriteToChannel, readCount);
+            }
+
+            // string message = $"{channel.GetType().Name}:  " +
+            //                  $"SingleReader: {options.SingleReader}, " +
+            //                  $"SingleWriter: {options.SingleWriter}, " +
+            //                  $"AllowSynchronousContinuations: {options.AllowSynchronousContinuations}, " +
+            //                  (boundedOptions == null ? "" : $"Mode: {boundedOptions.FullMode}, Capacity: {boundedOptions.Capacity},") +
+            //                  $"read: {readCount}, " +
+            //                  $"Max: {MaxNumberToWriteToChannel}";
+            // Console.WriteLine(message);
```

Comment:
Delete this comment?

Reply:
I was keeping it because it is handy to enable for more verbose output when trying to diagnose the execution. but I'll delete it anyway.